1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,74 @@
-function setAlarm() {}
+const display = document.querySelector("#timeRemaining");
+var audio = new Audio("alarmsound.mp3");
+audio.loop = true;
+let alarmTime = null;// alarme time and alarmSet for me
+let alarmtimeout = null;
+
+
+function updateTime(){
+  const date = new Date();
+
+  const hour = formatTime(date.getHours());
+  const ninutes = formatTime(date.getMinutes());
+  const seconds = formatTime(date.getSeconds());
+
+  display.innerHTML = `${hour} : ${ninutes} : ${seconds}`;
+}
+
+function formatTime(time){
+if (time < 10){
+  return "0" + time;
+}
+return time;
+}
+
+function setAlarmTime(value){
+alarmTime = value;
+
+}
+
+function setAlarm (){
+  if (alarmTime){
+    const current = new Date();
+    const timeToAlarm = new Date(alarmTime);
+
+    if (timeToAlarm > current){
+      const timeout = timeToAlarm.getTime() - current.getTime();
+      alarmtimeout = setTimeout(() => audio.play(), timeout);
+    alert("Set Alarm");
+    }
+  }
+}
+
+function clearAlarm(){
+  audio.pause();
+  if (alarmtimeout){
+    clearTimeout(alarmtimeout);
```

Comment:
This works kind of but it won't stop an existing call to the function, and as loop is set on the audio, if it plays, it'll keep playing forever until you stop it